data-prepper-plugins/otel-trace-group-prepper/src/test/java/com/amazon/dataprepper/plugins/prepper/oteltracegroup/OtelTraceGroupPrepperTests.java

```diff
@@ -0,0 +1,256 @@
+    private static final String TEST_TRACE_GROUP_2 = "/test_trace_group_2";
+    private static final String TEST_RAW_SPAN_COMPLETE_JSON_FILE_1 = "raw-span-complete-1.json";
+    private static final String TEST_RAW_SPAN_COMPLETE_JSON_FILE_2 = "raw-span-complete-2.json";
+    private static final String TEST_RAW_SPAN_MISSING_TRACE_GROUP_JSON_FILE_1 = "raw-span-missing-trace-group-1.json";
+    private static final String TEST_RAW_SPAN_MISSING_TRACE_GROUP_JSON_FILE_2 = "raw-span-missing-trace-group-2.json";
+    private static final int TEST_NUM_WORKERS = 2;
+
+    private MockedStatic<ConnectionConfiguration> connectionConfigurationMockedStatic;
+
+    private OtelTraceGroupPrepper otelTraceGroupPrepper;
+    private ExecutorService executorService;
+
+    @Mock
+    private ConnectionConfiguration connectionConfigurationMock;
+
+    @Mock
+    private RestHighLevelClient restHighLevelClient;
+
+    @Mock
+    private SearchResponse testSearchResponse;
+
+    @Mock
+    private SearchHits testSearchHits;
+
+    @Mock
+    private SearchHit testSearchHit1;
+
+    @Mock
+    private SearchHit testSearchHit2;
+
+    @Before
+    public void setUp() throws Exception{
+        connectionConfigurationMockedStatic = Mockito.mockStatic(ConnectionConfiguration.class);
+        connectionConfigurationMockedStatic.when(() -> ConnectionConfiguration.readConnectionConfiguration(any(PluginSetting.class)))
+                .thenReturn(connectionConfigurationMock);
+        when(connectionConfigurationMock.createClient()).thenReturn(restHighLevelClient);
+        when(restHighLevelClient.search(any(SearchRequest.class), any(RequestOptions.class))).thenReturn(testSearchResponse);
+        doNothing().when(restHighLevelClient).close();
+        when(testSearchResponse.getHits()).thenReturn(testSearchHits);
+        when(testSearchHits.getHits()).thenReturn(new SearchHit[] {testSearchHit1});
+        when(testSearchHit1.field("traceId")).thenReturn(new DocumentField("traceId", Collections.singletonList(TEST_TRACE_ID_1)));
+        when(testSearchHit1.field("traceGroup")).thenReturn(new DocumentField("traceGroup", Collections.singletonList(TEST_TRACE_GROUP_1)));
+        when(testSearchHit2.field("traceId")).thenReturn(new DocumentField("traceId", Collections.singletonList(TEST_TRACE_ID_2)));
+        when(testSearchHit2.field("traceGroup")).thenReturn(new DocumentField("traceGroup", Collections.singletonList(TEST_TRACE_GROUP_2)));
+        final PluginSetting testPluginSetting = new PluginSetting("otel_trace_group_prepper", new HashMap<>()) {{
+            setPipelineName("testPipelineName");
+        }};
+        otelTraceGroupPrepper = new OtelTraceGroupPrepper(testPluginSetting);
+        executorService = Executors.newFixedThreadPool(TEST_NUM_WORKERS);
+    }
+
+    @After
+    public void tearDown() {
+        otelTraceGroupPrepper.shutdown();
+        connectionConfigurationMockedStatic.close();
```

Comment:
Is this necessary?

Reply:
We encountered the mockstatic leakage issue in a previous PR. So close it here for safety.